src/ring_theory/adjoin/basic.lean

```diff
@@ -70,6 +70,28 @@ let S : subalgebra R A :=
 { carrier := p, mul_mem' := Hmul, add_mem' := Hadd, algebra_map_mem' := Halg } in
 adjoin_le (show s ≤ S, from Hs) h
 
+@[elab_as_eliminator]
+lemma adjoin_induction₂ {p : A → A → Prop} {a b : A} (ha : a ∈ adjoin R s) (hb : b ∈ adjoin R s)
```

Comment:
I think this could do with a brief docstring summary, as the argument list is so long!

Reply:
Fair enough!